### ISSUE TITLE
MissingSwitchDefault for switch exception (Java 14+)

### ISSUE THREAD

**Issue by sukolenvo** [checkstyle/checkstyle#11208]
I have read check documentation: https://checkstyle.org/config_coding.html#MissingSwitchDefault
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Main.java
#[[MAKE SURE THERE IS SUCCESSFUL COMPILATION]]

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
  "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name = "Checker">
  <module name="TreeWalker">
    <module name="MissingSwitchDefault"/>
  </module>
</module>


/var/tmp $ cat Main.java
import java.nio.file.AccessMode;

public class Main {

  private void createPlatform() {
    switch (AccessMode.READ) {
      case READ -> "1";
      case WRITE -> "2";
      case EXECUTE -> "3";
    };
  }
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-9.2.1-all.jar -c config.xml Main.java
Starting audit...
[ERROR] /home/gtx/Downloads/Main.java:7:12: switch without "default" clause. [MissingSwitchDefault]
Audit done.
Checkstyle ends with 1 errors.
```
**Expected**: no errors. Following code is very similar and does not produce any errors:
```java
  private void createPlatform() {
    String mode = switch (AccessMode.READ) {
      case READ -> "1";
      case WRITE -> "2";
      case EXECUTE -> "3";
    };
  }

  private String createPlatform() {
    return switch (AccessMode.READ) {
      case READ -> "1";
      case WRITE -> "2";
      case EXECUTE -> "3";
    };
  }
```
Given javac won't compile non-exhaustive switch, maybe `MissingSwitchDefault` check should be no-op for switch exceptions? Sample javac error:
```bash
/var/tmp $ cat Main.java 
import java.nio.file.AccessMode;

public class Main {

  private String createPlatform() {
    return switch (AccessMode.READ) {
      case WRITE -> "2";
      case EXECUTE -> "3";
    };
  }
}

/var/tmp $ javac Main.java
Main.java:6: error: the switch expression does not cover all possible input values
    return switch (AccessMode.READ) {
           ^
1 error
```
Related to #10463 and #10866

**Comment by sukolenvo** [comment]
Confused `->` with requirement of exhaustive `case`-s.
Thanks https://github.com/checkstyle/checkstyle/issues/10866#issuecomment-1014083535 for clarifying.

Closing.
